frontend/src/main/web/src/components/OverlayTrigger/index.jsx

```diff
@@ -0,0 +1,294 @@
+  // https://github.com/facebook/react/issues/4251
+  // for cases when the trigger is disabled and mouseOut/Over can
+  // cause flicker moving from one child element to another.
+  handleMouseOverOut = (handler, e) => {
+    let target = e.currentTarget
+    let related = e.relatedTarget || e.nativeEvent.toElement
+
+    if (!related || related !== target && !contains(target, related)) {
+      handler(e)
+    }
+  }
+  componentWillMount () {
+    this.handleMouseOver =
+      this.handleMouseOverOut.bind(null, this.handleDelayedShow)
+    this.handleMouseOut =
+      this.handleMouseOverOut.bind(null, this.handleDelayedHide)
+  }
+  componentDidMount () {
+    this._mountNode = document.createElement('div')
+    this.renderOverlay()
+  }
+  componentWillUnmount () {
+    ReactDOM.unmountComponentAtNode(this._mountNode)
+    this._mountNode = null
+    clearTimeout(this._hoverShowDelay)
+    clearTimeout(this._hoverHideDelay)
+  }
+  componentDidUpdate () {
+    if (this._mountNode) {
+      this.renderOverlay()
+    }
+  }
+  render () {
+    const trigger = React.Children.only(this.props.children)
+    const triggerProps = trigger.props
+
+    let props = {
+      'aria-describedby': this.props.overlay.props.id
+    }
+
+    // create in render otherwise owner is lost...
+    this._overlay = this.getOverlay()
+
+    props.onClick = createChainedFunction(
+      triggerProps.onClick,
+      this.props.onClick
+    )
+
+    if (isOneOf('click', this.props.trigger)) {
+      props.onClick = createChainedFunction(this.toggle, props.onClick)
+    }
+
+    if (isOneOf('hover', this.props.trigger)) {
+      warning(!(this.props.trigger === 'hover'),
+        `[zanata] Specifying only the "hover" trigger limits the
+        visibilty of the overlay to just mouse users. Consider also including
+        the "focus" trigger so that touch and keyboard only users can see
+        the overlay as well.`)
+
+      props.onMouseOver = createChainedFunction(
+        this.handleMouseOver,
+        this.props.onMouseOver,
+        triggerProps.onMouseOver
+      )
+      props.onMouseOut = createChainedFunction(
+        this.handleMouseOut,
+        this.props.onMouseOut,
+        triggerProps.onMouseOut
+      )
+    }
+
+    if (isOneOf('focus', this.props.trigger)) {
+      props.onFocus = createChainedFunction(
+        this.handleDelayedShow,
+        this.props.onFocus,
+        triggerProps.onFocus
+      )
+      props.onBlur = createChainedFunction(
+        this.handleDelayedHide,
+        this.props.onBlur,
+        triggerProps.onBlur
+      )
+    }
+
+    return cloneElement(
+      trigger,
+      props
+    )
+  }
+
+}
+
+OverlayTrigger.propTypes = {
+  ...Overlay.propTypes,
+   /**
+   * Specify which action or actions trigger Overlay visibility
+   */
+  trigger: PropTypes.oneOfType([
+    PropTypes.oneOf(['click', 'hover', 'focus']),
+    PropTypes.arrayOf(PropTypes.oneOf(['click', 'hover', 'focus']))
+  ]),
```

Comment:
It looks like a lot of extra complexity has been added to be able to specify a single trigger string or multiple trigger strings. We could just call it `triggers` and always take an array. It just means we have to use `['click']` instead of `'click'`, but I think that's much better than the extra complexity.